dask/tests/test_docs.py

```diff
@@ -0,0 +1,16 @@
+from pathlib import Path
+
+
+def test_development_guidelines_matches_ci():
+    """When the environment.yaml changes in CI, make sure to change it in the docs as well"""
+    root_dir = Path(__file__).parent.parent.parent
+
+    development_doc_file = root_dir / "docs" / "source" / "develop.rst"
+    ci_file = root_dir / ".github" / "workflows" / "additional.yml"
```

Comment:
We might also want to add `.github/workflows/upstream.yml` to ensure our `upstream` CI build is running against the latest support Python version too